Examples/Io/Root/src/RootAthenaDumpReader.cpp

```diff
@@ -161,28 +129,40 @@ RootAthenaDumpReader::RootAthenaDumpReader(
   m_inputchain->SetBranchAddress("CLnorm_y", CLnorm_y);
   m_inputchain->SetBranchAddress("CLnorm_z", CLnorm_z);
   m_inputchain->SetBranchAddress("CLlocal_cov", &CLlocal_cov);
-  m_inputchain->SetBranchAddress("nPartEVT", &nPartEVT);
-  m_inputchain->SetBranchAddress("Part_event_number", Part_event_number);
-  m_inputchain->SetBranchAddress("Part_barcode", Part_barcode);
-  m_inputchain->SetBranchAddress("Part_px", Part_px);
-  m_inputchain->SetBranchAddress("Part_py", Part_py);
-  m_inputchain->SetBranchAddress("Part_pz", Part_pz);
-  m_inputchain->SetBranchAddress("Part_pt", Part_pt);
-  m_inputchain->SetBranchAddress("Part_eta", Part_eta);
-  m_inputchain->SetBranchAddress("Part_vx", Part_vx);
-  m_inputchain->SetBranchAddress("Part_vy", Part_vy);
-  m_inputchain->SetBranchAddress("Part_vz", Part_vz);
-  m_inputchain->SetBranchAddress("Part_radius", Part_radius);
-  m_inputchain->SetBranchAddress("Part_status", Part_status);
-  m_inputchain->SetBranchAddress("Part_charge", Part_charge);
-  m_inputchain->SetBranchAddress("Part_pdg_id", Part_pdg_id);
-  m_inputchain->SetBranchAddress("Part_passed", Part_passed);
-  m_inputchain->SetBranchAddress("Part_vProdNin", Part_vProdNin);
-  m_inputchain->SetBranchAddress("Part_vProdNout", Part_vProdNout);
-  m_inputchain->SetBranchAddress("Part_vProdStatus", Part_vProdStatus);
-  m_inputchain->SetBranchAddress("Part_vProdBarcode", Part_vProdBarcode);
-  m_inputchain->SetBranchAddress("Part_vParentID", &Part_vParentID);
-  m_inputchain->SetBranchAddress("Part_vParentBarcode", &Part_vParentBarcode);
+  if (!m_cfg.noTruth) {
+    m_inputchain->SetBranchAddress("CLparticleLink_eventIndex",
+                                   &CLparticleLink_eventIndex);
+    m_inputchain->SetBranchAddress("CLparticleLink_barcode",
+                                   &CLparticleLink_barcode);
+    m_inputchain->SetBranchAddress("CLbarcodesLinked", &CLbarcodesLinked);
+    m_inputchain->SetBranchAddress("CLparticle_charge", &CLparticle_charge);
+  }
+
+  if (!m_cfg.noTruth) {
```

Comment:
is there a reason to separate these `if`s?

Reply:
Hmm I wanted to separate the groups...